drivers/trim_util.py

```diff
@@ -67,6 +67,17 @@ def to_xml(d):
     return dom.toxml()
 
 
+def _handle_discard_error(vg_name, sr_uuid, error):
+    if "BLKDISCARD ioctl failed: Operation not supported" not in error.reason:
+        err_msg = {
+            ERROR_CODE_KEY: 'TrimException',
+            ERROR_MSG_KEY: error.reason
+        }
+        return to_xml(err_msg)
+    else:
+        return str(True)
```

Comment:
Perhaps the right approach here would be to have an 'allowed operations' style check for trim, and if trim is not available on the underlying device then it could be reported ahead of time (and e.g. the GUI could grey out the option). We'd need to make trim a proper API instead of a plugin for that I think.

Reply:
As I said in the comment, disallowing it is the wrong action as a LUN which declares itself to have provisioning_mode == full might still (at the discretion of the SAN manufacturer, implementer) accept a TRIM/DISCARD request and do something useful. So, restricting this functionality only to LUNS which state they are thinly provisioned is wrong.

Reporting anything back to the user in the case of the SAN rejecting it is also wrong as that is the source of the current confusion from end-users which this change is addressing. Expecting the end-user to understand the complexities of when DISCARD is and is not supported by the SAN is significantly beyond what we can expect.